hera_sim/visibilities/vis_cpu.py

```diff
@@ -0,0 +1,328 @@
+        self.antpos = self.uvdata.get_ENU_antpos()[0].astype(self._real_dtype)
+        self.freqs = self.uvdata.freq_array[0]
+
+    @property
+    def lsts(self):
+        """
+        Sets LSTs from uvdata if not already set.
+
+        Returns
+        -------
+        array_like
+            LSTs of observations. Shape=(NTIMES,).
+        """
+        try:
+            return self.__lsts
+        except AttributeError:
+            self.__lsts = self.uvdata.lst_array[::self.uvdata.Nbls]
+
+            return self.__lsts
+
+    def validate(self):
+        """Checks for correct input format."""
+        super(VisCPU, self).validate()
+
+        # This one in particular requires that every baseline is used!
+        N = len(self.uvdata.antenna_numbers)
+        # N(N-1)/2 unique cross-correlations + N autocorrelations.
+        if len(self.uvdata.get_antpairs()) != N * (N + 1) / 2:
+            raise ValueError("VisCPU requires using every pair of antennas, "
+                             "but the UVData object does not comply.")
+
+        if (len(self.uvdata.data_array) != len(self.uvdata.get_antpairs())
+                * len(self.lsts)):
+            raise ValueError("VisCPU requires that every baseline uses the "
+                             "same LSTS.")
+
+    def get_beam_lm(self):
+        """
+        Obtain the beam pattern in (l,m) co-ordinates for each antenna.
+
+        Returns
+        -------
+        array_like
+            The beam pattern in (l,m) for each antenna.
+            Shape=(NANT, BM_PIX, BM_PIX).
+
+        Notes
+        -----
+            Due to using the verbatim :func:`vis_cpu` function, the beam
+            cube must have an entry for each antenna, which is a bit of
+            a waste of memory in some cases. If this is changed in the
+            future, this method can be modified to only return one
+            matrix for each beam.
+        """
+        return np.array([
+            conversions.uvbeam_to_lm(
+                self.beams[self.beam_ids[i]], self.freqs, self.bm_pix
+            ) for i in range(self.n_ant)
+        ])
+
+    def get_diffuse_crd_eq(self):
+        """
+        Calculate equatorial coords of HEALPix sky pixels (Cartesian).
+
+        Returns
+        -------
+        array_like of self._real_dtype
+            The equatorial co-ordinates of each pixel.
+            Shape=(12*NPIX^2, 3).
+        """
+        diffuse_eq = conversions.healpix_to_crd_eq(self.sky_intensity[0])
+        return diffuse_eq.astype(self._real_dtype)
+
+    def get_point_source_crd_eq(self):
+        """
+        Calculate approximate HEALPix map of point sources.
+
+        Returns
+        -------
+        array_like
+            equatorial coordinates of Healpix pixels, in Cartesian
+            system. Shape=(3, NPIX).
+        """
+        ra, dec = self.point_source_pos.T
+        return np.array([np.cos(ra)*np.cos(dec), np.cos(dec)*np.sin(ra),
```

Comment:
Again, might want to consider using `np.asarray` here instead.

Reply:
Done.